src/cmd/build.rs

```diff
@@ -195,56 +217,130 @@ impl BuildCommand {
             verbosity = Verbosity::Quiet;
         }
 
-        let args = ExecuteArgs {
-            manifest_path,
-            verbosity,
-            build_mode,
-            network,
-            build_artifact: self.build_artifact,
-            unstable_flags,
-            optimization_passes,
-            keep_debug_symbols: self.keep_debug_symbols,
-            skip_linting: self.skip_linting,
-            output_type,
-        };
+        let mut build_results = Vec::new();
 
-        execute(args)
+        match self.build_all {
+            true => {
+                let workspace_members = get_cargo_workspace_members(&manifest_path)?;
+                let mut args = ExecuteArgs {
+                    verbosity,
+                    build_mode,
+                    network,
+                    build_artifact: self.build_artifact,
+                    unstable_flags: unstable_flags.clone(),
+                    optimization_passes,
+                    keep_debug_symbols: self.keep_debug_symbols,
+                    skip_linting: self.skip_linting,
+                    output_type: output_type.clone(),
+                    ..Default::default()
+                };
+                for package_id in workspace_members {
+                    args.manifest_path =
+                        util::extract_subcontract_manifest_path(package_id)
+                            .expect("Error extracting package manifest path");
+                    build_results.push(execute(args.clone())?);
```

Comment:
We should look into adding an extra new line or seperator between contract builds. The output of `--all` looks a bit cluttered right now

```
 [5/6] Generating metadata
    Updating crates.io index
   Compiling metadata-gen v0.1.0 (/private/var/folders/g9/_vhwpplx0jv_07rh238bvjw80000gn/T/cargo-contract_IA34El/foo/.ink/metadata_gen)
    Finished release [optimized] target(s) in 0.76s
     Running `/private/tmp/flipper_workspace/target/ink/foo/release/metadata-gen ''`
 [6/6] Generating bundle
 [1/6] Checking ink! linting rules
    Updating git repository `https://github.com/paritytech/ink/`
Checking with toolchain `nightly-2022-06-30-aarch64-apple-darwin`
   Compiling proc-macro2 v1.0.43
   Compiling quote v1.0.21
   Compiling unicode-ident v1.0.3
```

Maybe even a counter of `Building contract [x/y]` could be added

Reply:
https://github.com/paritytech/cargo-contract/pull/647/commits/3807b06a641723db193e81e10fc4d1f91bc78d07